sphinx/templates/latex/longtable.tex_t

```diff
@@ -1,3 +1,6 @@
+<% if not table.caption and labels -%>
+\phantomsection<%= labels -%>
+<% endif -%>
 \begin{savenotes}\sphinxatlongtablestart\begin{longtable}
```

Comment:
the `\sphinxatlongtablestart` by default will do `\par` and add some vertical blank space and a page break can happen there. Thus it is possible that links will target bottom of previous page, if longable starts immediately at top of page. This is minor problem, and fixing it would complicate both template and sphinx.sty. Thus +1 to ignore it for time being.

Reply:
Is it better to put labels between `\sphinxatlongtablestart` and `\begin{longtable}`?

Reply:
This wouldn't help because the link targets are at location of `\phantomsection`, not at location of `\label` in LaTeX source.

Reply:
Sorry, what I meant is like following:
```
\begin{savenotes}\sphinxatlongtablestart\phantomsection\label{foobar}\begin{longtable}
```
If this minor change helps the splitting, I can change to that easily.

Reply:
This would not work. I tested this with success on top of your patch

```diff
diff --git a/sphinx/templates/latex/longtable.tex_t b/sphinx/templates/latex/longtable.tex_t
index 2c9e034b9..d182001d8 100644
--- a/sphinx/templates/latex/longtable.tex_t
+++ b/sphinx/templates/latex/longtable.tex_t
@@ -1,6 +1,3 @@
-<% if not table.caption and labels -%>
-\phantomsection<%= labels -%>
-<% endif -%>
 \begin{savenotes}\sphinxatlongtablestart\begin{longtable}
 <%- if table.align == 'center' -%>
   [c]
@@ -12,8 +9,10 @@
 <%= table.get_colspec() %>
 <%- if table.caption -%>
 \caption{<%= ''.join(table.caption) %>\strut}<%= labels %>\\*[\sphinxlongtablecapskipadjust]
-<% endif -%>
 \hline
+<%- elif labels -%>
+\hline\noalign{\phantomsection<%= labels %>}%
+<%- endif -%>
 <%= ''.join(table.header) %>
 \endfirsthead
 
```

Reply:
MEMO: the longtable increments table counter even if no caption, so captioned longtables may be "Table 1", "Table 3" if there is non-captioned one in-between. This is other issue.

Reply:
ah sorry my diff was swallowing `\hline` in case of neither caption nor labels, here is corrected one:

```diff
diff --git a/sphinx/templates/latex/longtable.tex_t b/sphinx/templates/latex/longtable.tex_t
index 2c9e034b9..60e0a54ee 100644
--- a/sphinx/templates/latex/longtable.tex_t
+++ b/sphinx/templates/latex/longtable.tex_t
@@ -1,6 +1,3 @@
-<% if not table.caption and labels -%>
-\phantomsection<%= labels -%>
-<% endif -%>
 \begin{savenotes}\sphinxatlongtablestart\begin{longtable}
 <%- if table.align == 'center' -%>
   [c]
@@ -12,8 +9,12 @@
 <%= table.get_colspec() %>
 <%- if table.caption -%>
 \caption{<%= ''.join(table.caption) %>\strut}<%= labels %>\\*[\sphinxlongtablecapskipadjust]
-<% endif -%>
 \hline
+<%- elif labels -%>
+\hline\noalign{\phantomsection<%= labels %>}%
+<%- else -%>
+\hline
+<%- endif -%>
 <%= ''.join(table.header) %>
 \endfirsthead
 
```

Reply:
just updated. thank you for your wisdom!

Reply:
sorry my diff did not include updates to root tests :(